From e56f57e0f9bc87dd0df668c4372495b837f07f4b Mon Sep 17 00:00:00 2001
From: Stelios Tsampas <loathingkernel@gmail.com>
Date: Mon, 27 Jun 2022 01:20:04 +0300
Subject: [PATCH] Add PROTON_USER_COMPAT_DATA environment variable

If set, stores proton prefixes in the Steam Library located in the HOME directory
---
 proton | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/proton b/proton
index 30e9ed7e..1bb49341 100755
--- a/proton
+++ b/proton
@@ -1625,6 +1625,15 @@ if __name__ == "__main__":
         log("No compat data path?")
         sys.exit(1)
 
+    if "PROTON_USER_COMPAT_DATA" in os.environ:
+        if nonzero(os.environ["PROTON_USER_COMPAT_DATA"]):
+            compatpath = os.environ["STEAM_COMPAT_DATA_PATH"].split('/')
+            os.environ["STEAM_COMPAT_DATA_PATH"] = "{}/{}/{}/{}".format(
+                os.environ["STEAM_COMPAT_CLIENT_INSTALL_PATH"],
+                compatpath[-3], compatpath[-2], compatpath[-1])
+            if not os.path.exists(os.environ["STEAM_COMPAT_DATA_PATH"]):
+                os.makedirs(os.environ["STEAM_COMPAT_DATA_PATH"])
+
     g_proton = Proton(os.path.dirname(sys.argv[0]))
 
     g_proton.cleanup_legacy_dist()
-- 
2.36.1

